src/background/storage.js

```diff
@@ -35,33 +36,30 @@ class Storage {
     try {
       let storagePersistNeeded = false;
       this.local = await browser.storage.local.get();
+      this.preferences = await browser.storage.sync.get();
       if (!Object.keys(this.local).length) {
         this.local = {
           tempContainerCounter: 0,
           tempContainers: {},
-          tabContainerMap: {},
-          preferences: this.preferencesDefault
+          tabContainerMap: {}
         };
         debug('storage empty, setting defaults');
         storagePersistNeeded = true;
       } else {
         debug('storage loaded', this.local);
       }
+
       // set preferences defaults if not present
-      if (!this.local.preferences) {
+      if (!Object.keys(this.preferences).length) {
+        this.preferences = {...this.preferencesDefault, ...this.local.preferences};
+        delete this.local.preferences;  // Migrate previous local preferences
         debug('no preferences found, setting defaults', this.preferencesDefault);
-        this.local.preferences = this.preferencesDefault;
         storagePersistNeeded = true;
-      } else {
-        Object.keys(this.preferencesDefault).map(key => {
-          if (this.local.preferences[key] === undefined) {
-            debug('preference not found, setting default', key, this.preferencesDefault[key]);
-            this.local.preferences[key] = this.preferencesDefault[key];
-            storagePersistNeeded = true;
-          }
-        });
       }
 
+      // Make sure to default missing preferences with default value
+      this.preferences = {...this.preferencesDefault, ...this.preferences};
+
       if (storagePersistNeeded) {
```

Comment:
When a new preferencesDefault key is added this will no longer automatically set to true and thus the new preference default will not be available on the options page since the options page accesses storage directly - [tho still local and not sync](https://github.com/stoically/firefox-add-on-temporary-containers/blob/master/options/options.js#L187).

Reply:
Good find 🥇 should we use storage there too or switch to sync?

Reply:
I'm not exactly sure what you mean - could you please elaborate?

Reply:
If you with "use storage" meant to require/import the `storage.js` in the options page - then I'd say that's not that easy and I don't see the point of doing that. Changing it to `sync` sounds good to me.

Reply:
I mean using the storage.js interface to save the preferences.

It is a bit sad to let the preference page know about the underlying storage but I understand it might not be trivial to do it.

Reply:
If you would want to abstract it away, the way to go would imho be sending a message to the background to request the storage. The same is already done when [saving the preferences](https://github.com/stoically/firefox-add-on-temporary-containers/blob/master/options/options.js#L22-L26).

Reply:
Yes that's what I had in mind, should I give it a try?

Reply:
If you like to, sure.